oak_restricted_kernel_sdk_proc_macro/src/lib.rs

```diff
@@ -18,6 +18,35 @@ use proc_macro::TokenStream;
 use quote::quote;
 use syn::{parse_macro_input, Item, ItemFn, Result};
 
+/// Marks a function as the entrypoint to an enclave app and sets up an conviences such an
+/// allocator, logger, panic handler.
+///
+/// This macro assumes that crates using it have declared the [`no_std`] and [`no_main`] attributes,
+/// and the [`alloc_error_handler`] unstable feature of the rust compiler.
+///
+/// [`no_std`]: <https://github.com/rust-lang/rust/issues/51540>
+/// [`no_main`]: <https://docs.rust-embedded.org/embedonomicon/smallest-no-std.html#the-code>
+/// [`alloc_error_handler`]: <https://github.com/rust-lang/rust/issues/51540>
+///
+/// # Examples
+///
+/// Filename: src/main.rs
+///
+/// ```
+/// #![no_std]
+/// #![no_main]
+/// #![feature(alloc_error_handler)]
+///
+/// extern crate alloc;
+///
+/// use oak_restricted_kernel_sdk::entrypoint;
+///
+/// #[entrypoint]
+/// fn start_enclave_app() -> ! {
+///     // business logic starts here
+///     /* ... */
+/// }
+/// ```
```

Comment:
interesting bit of trivia, when running tests, cargo compiles this! and throws an error

```
11 | fn start_enclave_app() -> ! {
   |    -----------------      ^ expected `!`, found `()`
```

Reply:
I was going to ask about that, because indeed it does not seem semantically correct Rust, and cargo test should reject it. Anyways you can fix that with a `loop {}` or similar.

Reply:
An unimplemented takes care of it. But we run into issues still since the examples declares no_std, which leads us down the rabbit hole of needing to set up the proper target to run the test with (I got that working with https://doc.rust-lang.org/cargo/reference/unstable.html#per-package-target), and then also setting up our own no_std test harness (that's where I gave up). 

Unfortunately I think we have to instruct the compiler to ignore this doc test for now.